### PR TITLE
storage: fix the integration test of BR

### PR DESCRIPTION
**Pull request by 3pointer**
Signed-off-by: 3pointer <luancheng@pingcap.com>

### What problem does this PR solve?

After #10132 #9706 merge into master, BR's integration test would be failed. This PR try to fix them.

### What is changed and how it works?

What's Changed:
1. Fix the error lost in s3 put object.
2. Keep the `changehost` logic as before.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

### Release note 
- No release note

### CODE REVIEW

File: components/cloud/aws/src/s3.rs [hunk 1]
```diff
@@ -283,21 +283,31 @@ impl<'client> S3Uploader<'client> {
...
+                }),
+        )
+        .await
+        {
+            Ok(output) => output?.upload_id.ok_or_else(|| {
+                RusotoError::ParseError(
+                    "missing upload-id from create_multipart_upload()".to_owned(),
+                )
+            }),
+            Err(_) => Err(RusotoError::ParseError("begin timeout".to_owned())),
```
Comment (lonng) [inline components/cloud/aws/src/s3.rs:309]: It's better to provide a more clear error message.

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Pull request merged**